sys-libs/db/db-5.3.28-r4.ebuild

```diff
@@ -0,0 +1,249 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit db flag-o-matic java-pkg-opt-2 autotools multilib-minimal toolchain-funcs
+
+#Number of official patches
+#PATCHNO=`echo ${PV}|sed -e "s,\(.*_p\)\([0-9]*\),\2,"`
+PATCHNO=${PV/*.*.*_p}
+if [[ ${PATCHNO} == "${PV}" ]] ; then
+	MY_PV=${PV}
+	MY_P=${P}
+	PATCHNO=0
+else
+	MY_PV=${PV/_p${PATCHNO}}
+	MY_P=${PN}-${MY_PV}
+fi
+
+S="${WORKDIR}/${MY_P}"
+BUILD_DIR="${S}/build${MULTIBUILD_VARIANT}"
+DESCRIPTION="Oracle Berkeley DB"
+HOMEPAGE="http://www.oracle.com/technetwork/database/database-technologies/berkeleydb/overview/index.html"
+SRC_URI="http://download.oracle.com/berkeley-db/${MY_P}.tar.gz"
+for (( i=1 ; i<=${PATCHNO} ; i++ )) ; do
+	export SRC_URI="${SRC_URI} http://www.oracle.com/technology/products/berkeley-db/db/update/${MY_PV}/patch.${MY_PV}.${i}"
+done
```

Comment:
couldn'- the patches here directly been added to `PATCHES` array ?

Reply:
Well, they do have to be here in order to download them. Could be directly added to the
patches array, but my goal is a minimal change to move to eapi7.

Reply:
just I just had a look, as you mentioned in #8314 "patches seem to be formatted to be -p0"...
so patches array is not a good idea